packages/purpose-process/src/utilities/errorMappers.ts

```diff
@@ -0,0 +1,19 @@
+import { constants } from "http2";
+import { ApiError, CommonErrorCodes } from "pagopa-interop-models";
+import { match } from "ts-pattern";
+import { ErrorCodes as LocalErrorCodes } from "../model/domain/errors.js";
+
+type ErrorCodes = LocalErrorCodes | CommonErrorCodes;
+
+const { HTTP_STATUS_INTERNAL_SERVER_ERROR, HTTP_STATUS_NOT_FOUND } = constants;
+
+export const getPurposeErrorMapper = (error: ApiError<ErrorCodes>): number =>
+  match(error.code)
+    .with(
+      "purposeNotFound",
+      "tenantNotFound",
+      "tenantKindNotFound",
+      () => HTTP_STATUS_NOT_FOUND
+    )
+    .with("eserviceNotFound", () => HTTP_STATUS_INTERNAL_SERVER_ERROR)
```

Comment:
No need to define the internal server error cases
```suggestion
```

Reply:
Updated